src/state/slices/dataset_slice.ts

```diff
@@ -7,27 +7,22 @@ import { CollectionSlice } from "./collection_slice";
 
 import Dataset from "../../colorizer/Dataset";
 
-type DatasetSliceState =
-  | {
-      datasetKey: null;
-      dataset: null;
-      featureKey: null;
-      track: null;
-      /** The key of the backdrop image set in the current dataset. `null` if there
-       * is no Dataset loaded or if the dataset does not have backdrops. */
-      backdropKey: null;
-    }
-  | {
-      datasetKey: string;
-      dataset: Dataset;
-      featureKey: string;
-      track: Track | null;
-      /** The key of the backdrop image set in the current dataset. `null` if there
-       * is no Dataset loaded or if the dataset does not have backdrops. */
-      backdropKey: string | null;
-    };
-
-type DatasetSliceActions = {
+export type DatasetSliceState = {
+  datasetKey: string | null;
+  dataset: Dataset | null;
+  featureKey: string | null;
+  track: Track | null;
+  /** The key of the backdrop image set in the current dataset. `null` if there
+   * is no Dataset loaded or if the dataset does not have backdrops. */
+  backdropKey: string | null;
+};
+
+export type DatasetSliceSerializableState = Pick<
+  DatasetSliceState,
+  "datasetKey" | "featureKey" | "track" | "backdropKey"
+>;
```

Comment:
I just learned through this PR what Pick is 😅, so I have a basic question. What makes you choose Pick vs Omit in a situation like this? Is it because you more specifically want to identify what parts you want instead of what parts you don't want?

Reply:
Yes, exactly! In this case, I want a developer (me) to have to manually add new properties that will be serialized by picking them, rather than having them be included in the type automatically. 

I _could_ get the same final type with `Omit`, but it would be a different development experience.

If I did `Omit<DatasetSliceState, "dataset">`, any new property would be marked as serializable automatically, which could potentially lead to bugs if I don't actually update the methods that serialize the slice.